File: user/src/main/kotlin/chatox/user/security/CurrentUser.kt
package chatox.user.security

import chatox.user.api.response.GlobalBanResponse
import chatox.user.api.response.UploadResponse
import chatox.user.domain.ImageUploadMetadata
import java.time.ZonedDateTime

data class CurrentUser(
        val id: String,
        val roles: List<String>,
        val firstName: String,
        val lastName: String?,
        val slug: String?,
        val avatar: UploadResponse<ImageUploadMetadata>?,
        val accountId: String,
        val bio: String?,
        val createdAt: ZonedDateTime,
        val dateOfBirth: ZonedDateTime?,
        val email: String?,
        val globalBan: GlobalBanResponse?,
        val externalAvatarUri: String?,
)

File: client/src/api/types/response/CurrentUser.ts
import {UserRole} from "./UserRole";
import {Upload} from "./Upload";
import {ImageUploadMetadata} from "./ImageUploadMetadata";
import {GlobalBan} from "./GlobalBan";

export interface CurrentUser {
    id: string,
    slug?: string,
    bio?: string,
    firstName: string,
    lastName?: string,
    avatarId?: string,
    avatar?: Upload<ImageUploadMetadata>,
    accountId: string,
    roles: UserRole[],
    createdAt: string,
    dateOfBirth?: string,
    email?: string,
    globalBan?: GlobalBan,
    externalAvatarUri?: string
}
